## `nixpkgs-review` result

Generated using [`nixpkgs-review-gha`](https://github.com/Defelo/nixpkgs-review-gha)

Command: `nixpkgs-review pr 420916`

Logs: https://github.com/GaetanLepage/nixpkgs-review-gha/actions/runs/16082062339


---
### `x86_64-linux` (sandbox = true)
<details>
  <summary>:white_check_mark: 40 packages built:</summary>
  <ul>
    <li>molbar (python313Packages.molbar)</li>
    <li>molbar.dist (python313Packages.molbar.dist)</li>
    <li>python312Packages.ase</li>
    <li>python312Packages.ase.dist</li>
    <li>python312Packages.boltztrap2</li>
    <li>python312Packages.boltztrap2.dist</li>
    <li>python312Packages.dscribe</li>
    <li>python312Packages.dscribe.dist</li>
    <li>python312Packages.gpaw</li>
    <li>python312Packages.gpaw.dist</li>
    <li>python312Packages.molbar</li>
    <li>python312Packages.molbar.dist</li>
    <li>python312Packages.nglview</li>
    <li>python312Packages.nglview.dist</li>
    <li>python312Packages.pymatgen</li>
    <li>python312Packages.pymatgen.dist</li>
    <li>python312Packages.pyscf</li>
    <li>python312Packages.pyscf.dist</li>
    <li>python312Packages.qcengine</li>
    <li>python312Packages.qcengine.dist</li>
    <li>python312Packages.simple-dftd3</li>
    <li>python312Packages.simple-dftd3.dist</li>
    <li>python312Packages.sumo</li>
    <li>python312Packages.sumo.dist</li>
    <li>python313Packages.ase</li>
    <li>python313Packages.ase.dist</li>
    <li>python313Packages.boltztrap2</li>
    <li>python313Packages.boltztrap2.dist</li>
    <li>python313Packages.dscribe</li>
    <li>python313Packages.dscribe.dist</li>
    <li>python313Packages.gpaw</li>
    <li>python313Packages.gpaw.dist</li>
    <li>python313Packages.nglview</li>
    <li>python313Packages.nglview.dist</li>
    <li>python313Packages.pyscf</li>
    <li>python313Packages.pyscf.dist</li>
    <li>python313Packages.qcengine</li>
    <li>python313Packages.qcengine.dist</li>
    <li>python313Packages.simple-dftd3</li>
    <li>python313Packages.simple-dftd3.dist</li>
  </ul>
</details>


---
### `aarch64-linux` (sandbox = true)
<details>
  <summary>:white_check_mark: 32 packages built:</summary>
  <ul>
    <li>molbar (python313Packages.molbar)</li>
    <li>molbar.dist (python313Packages.molbar.dist)</li>
    <li>python312Packages.ase</li>
    <li>python312Packages.ase.dist</li>
    <li>python312Packages.boltztrap2</li>
    <li>python312Packages.boltztrap2.dist</li>
    <li>python312Packages.dscribe</li>
    <li>python312Packages.dscribe.dist</li>
    <li>python312Packages.gpaw</li>
    <li>python312Packages.gpaw.dist</li>
    <li>python312Packages.molbar</li>
    <li>python312Packages.molbar.dist</li>
    <li>python312Packages.nglview</li>
    <li>python312Packages.nglview.dist</li>
    <li>python312Packages.pymatgen</li>
    <li>python312Packages.pymatgen.dist</li>
    <li>python312Packages.qcengine</li>
    <li>python312Packages.qcengine.dist</li>
    <li>python312Packages.sumo</li>
    <li>python312Packages.sumo.dist</li>
    <li>python313Packages.ase</li>
    <li>python313Packages.ase.dist</li>
    <li>python313Packages.boltztrap2</li>
    <li>python313Packages.boltztrap2.dist</li>
    <li>python313Packages.dscribe</li>
    <li>python313Packages.dscribe.dist</li>
    <li>python313Packages.gpaw</li>
    <li>python313Packages.gpaw.dist</li>
    <li>python313Packages.nglview</li>
    <li>python313Packages.nglview.dist</li>
    <li>python313Packages.qcengine</li>
    <li>python313Packages.qcengine.dist</li>
  </ul>
</details>


---
### `x86_64-darwin` (sandbox = true)
<details>
  <summary>:white_check_mark: 28 packages built:</summary>
  <ul>
    <li>molbar (python313Packages.molbar)</li>
    <li>molbar.dist (python313Packages.molbar.dist)</li>
    <li>python312Packages.ase</li>
    <li>python312Packages.ase.dist</li>
    <li>python312Packages.boltztrap2</li>
    <li>python312Packages.boltztrap2.dist</li>
    <li>python312Packages.dscribe</li>
    <li>python312Packages.dscribe.dist</li>
    <li>python312Packages.molbar</li>
    <li>python312Packages.molbar.dist</li>
    <li>python312Packages.nglview</li>
    <li>python312Packages.nglview.dist</li>
    <li>python312Packages.pymatgen</li>
    <li>python312Packages.pymatgen.dist</li>
    <li>python312Packages.pyscf</li>
    <li>python312Packages.pyscf.dist</li>
    <li>python312Packages.sumo</li>
    <li>python312Packages.sumo.dist</li>
    <li>python313Packages.ase</li>
    <li>python313Packages.ase.dist</li>
    <li>python313Packages.boltztrap2</li>
    <li>python313Packages.boltztrap2.dist</li>
    <li>python313Packages.dscribe</li>
    <li>python313Packages.dscribe.dist</li>
    <li>python313Packages.nglview</li>
    <li>python313Packages.nglview.dist</li>
    <li>python313Packages.pyscf</li>
    <li>python313Packages.pyscf.dist</li>
  </ul>
</details>


---
### `aarch64-darwin` (sandbox = true)
<details>
  <summary>:x: 8 packages failed to build:</summary>
  <ul>
    <li>python312Packages.gpaw</li>
    <li>python312Packages.gpaw.dist</li>
    <li>python312Packages.pyscf</li>
    <li>python312Packages.pyscf.dist</li>
    <li>python313Packages.gpaw</li>
    <li>python313Packages.gpaw.dist</li>
    <li>python313Packages.pyscf</li>
    <li>python313Packages.pyscf.dist</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 24 packages built:</summary>
  <ul>
    <li>molbar (python313Packages.molbar)</li>
    <li>molbar.dist (python313Packages.molbar.dist)</li>
    <li>python312Packages.ase</li>
    <li>python312Packages.ase.dist</li>
    <li>python312Packages.boltztrap2</li>
    <li>python312Packages.boltztrap2.dist</li>
    <li>python312Packages.dscribe</li>
    <li>python312Packages.dscribe.dist</li>
    <li>python312Packages.molbar</li>
    <li>python312Packages.molbar.dist</li>
    <li>python312Packages.nglview</li>
    <li>python312Packages.nglview.dist</li>
    <li>python312Packages.pymatgen</li>
    <li>python312Packages.pymatgen.dist</li>
    <li>python312Packages.sumo</li>
    <li>python312Packages.sumo.dist</li>
    <li>python313Packages.ase</li>
    <li>python313Packages.ase.dist</li>
    <li>python313Packages.boltztrap2</li>
    <li>python313Packages.boltztrap2.dist</li>
    <li>python313Packages.dscribe</li>
    <li>python313Packages.dscribe.dist</li>
    <li>python313Packages.nglview</li>
    <li>python313Packages.nglview.dist</li>
  </ul>
</details>

---
<details>
<summary>Error logs: `aarch64-darwin`</summary>
<details>
<summary>python312Packages.gpaw</summary>
<pre>ld: warning: directory not found for option '-L/nix/store/eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee-mpdecimal-4.0.1/lib'
ld: warning: directory not found for option '-L/nix/store/eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee-expat-2.7.1/lib'
ld: warning: directory not found for option '-L/nix/store/gsbyj7v3rzg441j8wrqgilijg9sz0g90-tzdata-2025b/lib'
ld: warning: directory not found for option '-L/nix/store/eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee-gdbm-1.25-lib/lib'
ld: warning: directory not found for option '-L/nix/store/eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee-readline-8.2p13/lib'
ld: warning: directory not found for option '-L/nix/store/eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee-bzip2-1.0.8/lib'
ld: warning: directory not found for option '-L/nix/store/eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee-libffi-39/lib'
ld: warning: directory not found for option '-L/nix/store/eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee-ncurses-6.5/lib'
ld: warning: directory not found for option '-L/nix/store/eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee-openssl-3.4.1/lib'
ld: warning: directory not found for option '-L/nix/store/eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee-sqlite-3.50.1/lib'
ld: warning: directory not found for option '-L/nix/store/eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee-xz-5.8.1/lib'
ld: warning: directory not found for option '-L/nix/store/eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee-zlib-1.3.1/lib'
ld: warning: directory not found for option '-L/nix/store/eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee-mpdecimal-4.0.1/lib'
ld: warning: directory not found for option '-L/nix/store/eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee-expat-2.7.1/lib'
ld: warning: directory not found for option '-L/nix/store/gsbyj7v3rzg441j8wrqgilijg9sz0g90-tzdata-2025b/lib'
ld: warning: directory not found for option '-L/nix/store/eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee-gdbm-1.25-lib/lib'
ld: warning: directory not found for option '-L/nix/store/eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee-readline-8.2p13/lib'
ld: unknown option: -rpath=/nix/store/hyrnwk5m63jld5nk082mpas4xkcwr8yl-libxc-6.2.2//lib
clang: error: linker command failed with exit code 1 (use -v to see invocation)
error: command '/nix/store/5xcm9yg78vmrlihynm05m9wsnb4lnndi-openmpi-5.0.6-dev/bin/mpicc' failed with exit code 1</pre>
</details>
<details>
<summary>python312Packages.pyscf</summary>
<pre>  /private/tmp/nix-build-python3.12-pyscf-2.9.0.drv-0/source/pyscf/solvent/test/test_ddcosmo.py:304: DeprecationWarning: `scipy.special.sph_harm` is deprecated as of SciPy 1.15.0 and will be removed in SciPy 1.17.0. Please use `scipy.special.sph_harm_y` instead.
    ylm[l] = scipy.special.sph_harm(0, l, theta, varphi).real

pyscf/solvent/test/test_ddcosmo.py: 2100 warnings
  /private/tmp/nix-build-python3.12-pyscf-2.9.0.drv-0/source/pyscf/solvent/test/test_ddcosmo.py:306: DeprecationWarning: `scipy.special.sph_harm` is deprecated as of SciPy 1.15.0 and will be removed in SciPy 1.17.0. Please use `scipy.special.sph_harm_y` instead.
    f1 = scipy.special.sph_harm(-m, l, theta, varphi)

pyscf/solvent/test/test_ddcosmo.py: 2100 warnings
  /private/tmp/nix-build-python3.12-pyscf-2.9.0.drv-0/source/pyscf/solvent/test/test_ddcosmo.py:307: DeprecationWarning: `scipy.special.sph_harm` is deprecated as of SciPy 1.15.0 and will be removed in SciPy 1.17.0. Please use `scipy.special.sph_harm_y` instead.
    f2 = scipy.special.sph_harm( m, l, theta, varphi)

pyscf/tools/test/test_fcidump.py::KnownValues::test_to_scf_with_symmetry
  /private/tmp/nix-build-python3.12-pyscf-2.9.0.drv-0/source/pyscf/lib/misc.py:648: UserWarning: method set() only supports keyword arguments.
  Arguments <pyscf.gto.mole.Mole object at 0x11f7b0890> are ignored.
    warnings.warn('method set() only supports keyword arguments.\n'

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
=========================== short test summary info ============================
FAILED pyscf/grad/test/test_hf.py::KnownValues::test_rks_scanner - numpy.linalg.LinAlgError: Internal Error.
= 1 failed, 2417 passed, 47 skipped, 327 deselected, 145797 warnings in 1291.75s (0:21:31) =</pre>
</details>
<details>
<summary>python313Packages.pyscf</summary>
<pre>  /private/tmp/nix-build-python3.13-pyscf-2.9.0.drv-0/source/pyscf/solvent/test/test_ddcosmo.py:304: DeprecationWarning: `scipy.special.sph_harm` is deprecated as of SciPy 1.15.0 and will be removed in SciPy 1.17.0. Please use `scipy.special.sph_harm_y` instead.
    ylm[l] = scipy.special.sph_harm(0, l, theta, varphi).real

pyscf/solvent/test/test_ddcosmo.py: 2100 warnings
  /private/tmp/nix-build-python3.13-pyscf-2.9.0.drv-0/source/pyscf/solvent/test/test_ddcosmo.py:306: DeprecationWarning: `scipy.special.sph_harm` is deprecated as of SciPy 1.15.0 and will be removed in SciPy 1.17.0. Please use `scipy.special.sph_harm_y` instead.
    f1 = scipy.special.sph_harm(-m, l, theta, varphi)

pyscf/solvent/test/test_ddcosmo.py: 2100 warnings
  /private/tmp/nix-build-python3.13-pyscf-2.9.0.drv-0/source/pyscf/solvent/test/test_ddcosmo.py:307: DeprecationWarning: `scipy.special.sph_harm` is deprecated as of SciPy 1.15.0 and will be removed in SciPy 1.17.0. Please use `scipy.special.sph_harm_y` instead.
    f2 = scipy.special.sph_harm( m, l, theta, varphi)

pyscf/tools/test/test_fcidump.py::KnownValues::test_to_scf_with_symmetry
  /private/tmp/nix-build-python3.13-pyscf-2.9.0.drv-0/source/pyscf/lib/misc.py:648: UserWarning: method set() only supports keyword arguments.
  Arguments <pyscf.gto.mole.Mole object at 0x1178e0190> are ignored.
    warnings.warn('method set() only supports keyword arguments.\n'

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
=========================== short test summary info ============================
FAILED pyscf/grad/test/test_hf.py::KnownValues::test_rks_scanner - numpy.linalg.LinAlgError: Internal Error.
= 1 failed, 2417 passed, 47 skipped, 327 deselected, 145797 warnings in 1479.29s (0:24:39) =</pre>
</details>
</details>

--- a/pkgs/development/python-modules/pyscf/coerce-numpy-to-int.patch
+++ b/pkgs/development/python-modules/pyscf/coerce-numpy-to-int.patch
@@ -1,0 +1,19 @@
+commit 6df405512d33d432bb45553ddcffbc70edec51f6
+Author: Lein Matsumaru <applePrincess@applePrincess.io>
+Date:   Sun Jun 29 12:59:57 2025 +0000
+
+    Coerce int
+
+diff --git a/pyscf/ci/gcisd.py b/pyscf/ci/gcisd.py
+index d58e0364c..050f83962 100644
+--- a/pyscf/ci/gcisd.py
++++ b/pyscf/ci/gcisd.py
+@@ -197,7 +197,7 @@ def from_fcivec(ci0, nelec, orbspin, frozen=None):
+              numpy.count_nonzero(orbspin[:nelec] == 1))
+     ucisdvec = ucisd.from_fcivec(ci0, norb//2, nelec, frozen)
+     nocc = numpy.count_nonzero(~frozen_mask[:sum(nelec)])
+-    return from_ucisdvec(ucisdvec, nocc, orbspin[~frozen_mask])
++    return from_ucisdvec(ucisdvec, int(nocc), orbspin[~frozen_mask])
+ 
+ 
+ def make_rdm1(myci, civec=None, nmo=None, nocc=None, ao_repr=False):

--- a/pkgs/development/python-modules/pyscf/default.nix
+++ b/pkgs/development/python-modules/pyscf/default.nix
@@ -2,15 +2,23 @@
   buildPythonPackage,
   lib,
   fetchFromGitHub,
+
+  # build-sysetm
   cmake,
+
+  # build inputs
   blas,
   libcint,
   libxc,
   xcfun,
+
+  # dependencies
   cppe,
   h5py,
   numpy,
   scipy,
+
+  # tests
   pytestCheckHook,
 }:
 
@@ -25,6 +33,12 @@ buildPythonPackage rec {
     tag = "v${version}";
     hash = "sha256-UTeZXlNuSWDOcBRVbUUWJ3mQnZZQr17aTw6rRA5DRNI=";
   };
+
+  patches = [
+    # Converts numpy.int64 to int where necessary.
+    # Upstream issue: https://github.com/pyscf/pyscf/issues/2878
+    ./coerce-numpy-to-int.patch
+  ];
 
   # setup.py calls Cmake and passes the arguments in CMAKE_CONFIGURE_ARGS to cmake.
   build-system = [ cmake ];
@@ -60,6 +74,7 @@ buildPythonPackage rec {
 
   # Numerically slightly off tests
   disabledTests = [
+    "test_rdm_trace"
     "test_tdhf_singlet"
     "test_ab_hf"
     "test_ea"
@@ -99,14 +114,15 @@ buildPythonPackage rec {
     "--ignore-glob=pyscf/grad/test/test_casscf.py"
   ];
 
-  meta = with lib; {
+  meta = {
     description = "Python-based simulations of chemistry framework";
     homepage = "https://github.com/pyscf/pyscf";
-    license = licenses.asl20;
+    license = lib.licenses.asl20;
     platforms = [
       "x86_64-linux"
       "x86_64-darwin"
+      "aarch64-darwin"
     ];
-    maintainers = [ maintainers.sheepforce ];
+    maintainers = [ lib.maintainers.sheepforce ];
   };
 }

